parachains/runtimes/bridge-hubs/test-utils/src/test_cases.rs

```diff
@@ -392,123 +347,586 @@ pub fn message_dispatch_routing_works<
-		$runtime_para_id:expr,
-		$sibling_parachain_id:expr,
-		$unwrap_cumulus_pallet_parachain_system_event:expr,
-		$unwrap_cumulus_pallet_xcmp_queue_event:expr,
-		$expected_lane_id:expr
-	) => {
-		#[test]
-		fn message_dispatch_routing_works() {
-			$crate::test_cases::message_dispatch_routing_works::<
-				$runtime,
-				$xcm_config,
-				$hrmp_channel_opener,
-				$pallet_bridge_messages_instance,
-				$runtime_network,
-				$bridged_network
-			>(
-				$collator_session_key,
-				$runtime_para_id,
-				$sibling_parachain_id,
-				$unwrap_cumulus_pallet_parachain_system_event,
-				$unwrap_cumulus_pallet_xcmp_queue_event,
-				$expected_lane_id,
+/// Test-case makes sure that Runtime can dispatch XCM messages submitted by relayer,
+/// with proofs (finality, para heads, message) independently submitted.
+pub fn relayed_incoming_message_works<Runtime, XcmConfig, HrmpChannelOpener, GPI, PPI, MPI, MB>(
+	collator_session_key: CollatorSessionKeys<Runtime>,
+	runtime_para_id: u32,
+	bridged_para_id: u32,
+	sibling_parachain_id: u32,
+	local_relay_chain_id: NetworkId,
+	lane_id: LaneId,
+) where
+	Runtime: frame_system::Config
+	+ pallet_balances::Config
+	+ pallet_session::Config
+	+ pallet_xcm::Config
+	+ parachain_info::Config
+	+ pallet_collator_selection::Config
+	+ cumulus_pallet_dmp_queue::Config
+	+ cumulus_pallet_parachain_system::Config
+	+ cumulus_pallet_xcmp_queue::Config
+	+ pallet_bridge_grandpa::Config<GPI>
+	+ pallet_bridge_parachains::Config<PPI>
+	+ pallet_bridge_messages::Config<MPI, InboundPayload = XcmAsPlainPayload>,
+	GPI: 'static,
+	PPI: 'static,
+	MPI: 'static,
+	MB: MessageBridge,
+	<MB as MessageBridge>::BridgedChain: Send + Sync + 'static,
+	UnderlyingChainOf<MessageBridgedChain<MB>>: bp_runtime::Chain<Hash = ParaHash> + Parachain,
+	XcmConfig: xcm_executor::Config,
+	HrmpChannelOpener: frame_support::inherent::ProvideInherent<
+		Call = cumulus_pallet_parachain_system::Call<Runtime>,
+	>,
+	ValidatorIdOf<Runtime>: From<AccountIdOf<Runtime>>,
+	<<Runtime as pallet_bridge_messages::Config<MPI>>::SourceHeaderChain as SourceHeaderChain>::MessagesProof: From<FromBridgedChainMessagesProof<ParaHash>>,
+	<<Runtime as pallet_bridge_grandpa::Config<GPI>>::BridgedChain as bp_runtime::Chain>::Hash: From<ParaHash>,
+	ParaHash: From<<<Runtime as pallet_bridge_grandpa::Config<GPI>>::BridgedChain as bp_runtime::Chain>::Hash>,
+	AccountIdOf<Runtime>: From<sp_core::sr25519::Public>,
+	<Runtime as pallet_bridge_messages::Config<MPI>>::InboundRelayer: From<AccountId32>,
+{
+	assert_ne!(runtime_para_id, sibling_parachain_id);
+	assert_ne!(runtime_para_id, bridged_para_id);
+
+	ExtBuilder::<Runtime>::default()
+		.with_collators(collator_session_key.collators())
+		.with_session_keys(collator_session_key.session_keys())
+		.with_safe_xcm_version(XCM_VERSION)
+		.with_para_id(runtime_para_id.into())
+		.with_tracing()
+		.build()
+		.execute_with(|| {
+			frame_system::Pallet::<Runtime>::set_block_number(1u32.into());
```

Comment:
I guess we dont need this two calls (set_block_number/reset_events),
at least `set_block_number(1)` (which enables events) is done in [ExtBuilder::build](https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/assets/test-utils/src/lib.rs#L156)